xline/src/main.rs

```diff
@@ -229,25 +232,26 @@ struct ServerArgs {
 
 impl From<ServerArgs> for XlineServerConfig {
     fn from(args: ServerArgs) -> Self {
-        let Ok(curp_config) = CurpConfigBuilder::default()
-            .heartbeat_interval(args.heartbeat_interval
-                .unwrap_or_else(default_heartbeat_interval))
-            .wait_synced_timeout(args.server_wait_synced_timeout
-                .unwrap_or_else(default_server_wait_synced_timeout))
-            .retry_timeout(args.retry_timeout.unwrap_or_else(default_retry_timeout))
-            .rpc_timeout(args.rpc_timeout.unwrap_or_else(default_rpc_timeout))
-            .batch_timeout(args.batch_timeout.unwrap_or_else(default_batch_timeout))
-            .batch_max_size(args.batch_max_size.unwrap_or_else(default_batch_max_size))
-            .follower_timeout_ticks(args.follower_timeout_ticks)
-            .candidate_timeout_ticks(args.candidate_timeout_ticks)
-            .data_dir(args.curp_dir.unwrap_or_else(|| {
-                    let mut path = args.data_dir.clone();
-                    path.push("curp");
-                    path
-                }))
-            .gc_interval(args.gc_interval.unwrap_or_else(default_gc_interval))
-            .cmd_workers(args.cmd_workers)
-            .build() else {unreachable!()};
+        let curp_config = CurpConfig {
```

Comment:
I prefer using `CurpConfigBuilder` here. If we have implemented a builder pattern for the `CurpCofnig`, then why should we construct a `CurpConfig` object in different ways?

Reply:
Fixed